// streaming-availability/dist/models/Addon.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddonToJSON = exports.AddonFromJSONTyped = exports.AddonFromJSON = exports.instanceOfAddon = void 0;
var ImageSet_1 = require("./ImageSet");
/**
 * Check if a given object implements the Addon interface.
 */
function instanceOfAddon(value) {
    var isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "displayName" in value;
    isInstance = isInstance && "homePage" in value;
    isInstance = isInstance && "themeColorCode" in value;
    isInstance = isInstance && "image" in value;
    isInstance = isInstance && "images" in value;
    return isInstance;
}
exports.instanceOfAddon = instanceOfAddon;
function AddonFromJSON(json) {
    return AddonFromJSONTyped(json, false);
}
exports.AddonFromJSON = AddonFromJSON;
function AddonFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': json['id'],
        'displayName': json['displayName'],
        'homePage': json['homePage'],
        'themeColorCode': json['themeColorCode'],
        'image': json['image'],
        'images': (0, ImageSet_1.ImageSetFromJSON)(json['images']),
    };
}
exports.AddonFromJSONTyped = AddonFromJSONTyped;
function AddonToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'displayName': value.displayName,
        'homePage': value.homePage,
        'themeColorCode': value.themeColorCode,
        'image': value.image,
        'images': (0, ImageSet_1.ImageSetToJSON)(value.images),
    };
}
exports.AddonToJSON = AddonToJSON;
