// streaming-availability/dist/models/ChangeSet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeSetToJSON = exports.ChangeSetFromJSONTyped = exports.ChangeSetFromJSON = exports.instanceOfChangeSet = void 0;
var Change_1 = require("./Change");
var Show_1 = require("./Show");
/**
 * Check if a given object implements the ChangeSet interface.
 */
function instanceOfChangeSet(value) {
    var isInstance = true;
    isInstance = isInstance && "changes" in value;
    isInstance = isInstance && "show" in value;
    return isInstance;
}
exports.instanceOfChangeSet = instanceOfChangeSet;
function ChangeSetFromJSON(json) {
    return ChangeSetFromJSONTyped(json, false);
}
exports.ChangeSetFromJSON = ChangeSetFromJSON;
function ChangeSetFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'changes': (json['changes'].map(Change_1.ChangeFromJSON)),
        'show': (0, Show_1.ShowFromJSON)(json['show']),
    };
}
exports.ChangeSetFromJSONTyped = ChangeSetFromJSONTyped;
function ChangeSetToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'changes': (value.changes.map(Change_1.ChangeToJSON)),
        'show': (0, Show_1.ShowToJSON)(value.show),
    };
}
exports.ChangeSetToJSON = ChangeSetToJSON;
