// streaming-availability/dist/models/ChangesResponseSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangesResponseSchemaToJSON = exports.ChangesResponseSchemaFromJSONTyped = exports.ChangesResponseSchemaFromJSON = exports.instanceOfChangesResponseSchema = void 0;
var ChangeSet_1 = require("./ChangeSet");
/**
 * Check if a given object implements the ChangesResponseSchema interface.
 */
function instanceOfChangesResponseSchema(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    isInstance = isInstance && "hasMore" in value;
    isInstance = isInstance && "nextCursor" in value;
    return isInstance;
}
exports.instanceOfChangesResponseSchema = instanceOfChangesResponseSchema;
function ChangesResponseSchemaFromJSON(json) {
    return ChangesResponseSchemaFromJSONTyped(json, false);
}
exports.ChangesResponseSchemaFromJSON = ChangesResponseSchemaFromJSON;
function ChangesResponseSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (json['result'].map(ChangeSet_1.ChangeSetFromJSON)),
        'hasMore': json['hasMore'],
        'nextCursor': json['nextCursor'],
    };
}
exports.ChangesResponseSchemaFromJSONTyped = ChangesResponseSchemaFromJSONTyped;
function ChangesResponseSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (value.result.map(ChangeSet_1.ChangeSetToJSON)),
        'hasMore': value.hasMore,
        'nextCursor': value.nextCursor,
    };
}
exports.ChangesResponseSchemaToJSON = ChangesResponseSchemaToJSON;
