// streaming-availability/dist/models/CountriesResponseSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CountriesResponseSchemaToJSON = exports.CountriesResponseSchemaFromJSONTyped = exports.CountriesResponseSchemaFromJSON = exports.instanceOfCountriesResponseSchema = void 0;
var runtime_1 = require("../runtime");
var Country_1 = require("./Country");
/**
 * Check if a given object implements the CountriesResponseSchema interface.
 */
function instanceOfCountriesResponseSchema(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfCountriesResponseSchema = instanceOfCountriesResponseSchema;
function CountriesResponseSchemaFromJSON(json) {
    return CountriesResponseSchemaFromJSONTyped(json, false);
}
exports.CountriesResponseSchemaFromJSON = CountriesResponseSchemaFromJSON;
function CountriesResponseSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': ((0, runtime_1.mapValues)(json['result'], Country_1.CountryFromJSON)),
    };
}
exports.CountriesResponseSchemaFromJSONTyped = CountriesResponseSchemaFromJSONTyped;
function CountriesResponseSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': ((0, runtime_1.mapValues)(value.result, Country_1.CountryToJSON)),
    };
}
exports.CountriesResponseSchemaToJSON = CountriesResponseSchemaToJSON;
