// streaming-availability/dist/models/Episode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EpisodeToJSON = exports.EpisodeFromJSONTyped = exports.EpisodeFromJSON = exports.instanceOfEpisode = exports.EpisodeTypeEnum = void 0;
/**
 * @export
 */
exports.EpisodeTypeEnum = {
    Episode: 'episode'
};
/**
 * Check if a given object implements the Episode interface.
 */
function instanceOfEpisode(value) {
    var isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "title" in value;
    isInstance = isInstance && "year" in value;
    isInstance = isInstance && "streamingInfo" in value;
    return isInstance;
}
exports.instanceOfEpisode = instanceOfEpisode;
function EpisodeFromJSON(json) {
    return EpisodeFromJSONTyped(json, false);
}
exports.EpisodeFromJSON = EpisodeFromJSON;
function EpisodeFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'type': json['type'],
        'title': json['title'],
        'year': json['year'],
        'streamingInfo': json['streamingInfo'],
    };
}
exports.EpisodeFromJSONTyped = EpisodeFromJSONTyped;
function EpisodeToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'title': value.title,
        'year': value.year,
        'streamingInfo': value.streamingInfo,
    };
}
exports.EpisodeToJSON = EpisodeToJSON;
