// streaming-availability/dist/models/GenresResponseSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenresResponseSchemaToJSON = exports.GenresResponseSchemaFromJSONTyped = exports.GenresResponseSchemaFromJSON = exports.instanceOfGenresResponseSchema = void 0;
/**
 * Check if a given object implements the GenresResponseSchema interface.
 */
function instanceOfGenresResponseSchema(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfGenresResponseSchema = instanceOfGenresResponseSchema;
function GenresResponseSchemaFromJSON(json) {
    return GenresResponseSchemaFromJSONTyped(json, false);
}
exports.GenresResponseSchemaFromJSON = GenresResponseSchemaFromJSON;
function GenresResponseSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': json['result'],
    };
}
exports.GenresResponseSchemaFromJSONTyped = GenresResponseSchemaFromJSONTyped;
function GenresResponseSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': value.result,
    };
}
exports.GenresResponseSchemaToJSON = GenresResponseSchemaToJSON;
