// streaming-availability/dist/models/GetResponseSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetResponseSchemaToJSON = exports.GetResponseSchemaFromJSONTyped = exports.GetResponseSchemaFromJSON = exports.instanceOfGetResponseSchema = void 0;
var Show_1 = require("./Show");
/**
 * Check if a given object implements the GetResponseSchema interface.
 */
function instanceOfGetResponseSchema(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfGetResponseSchema = instanceOfGetResponseSchema;
function GetResponseSchemaFromJSON(json) {
    return GetResponseSchemaFromJSONTyped(json, false);
}
exports.GetResponseSchemaFromJSON = GetResponseSchemaFromJSON;
function GetResponseSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (0, Show_1.ShowFromJSON)(json['result']),
    };
}
exports.GetResponseSchemaFromJSONTyped = GetResponseSchemaFromJSONTyped;
function GetResponseSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (0, Show_1.ShowToJSON)(value.result),
    };
}
exports.GetResponseSchemaToJSON = GetResponseSchemaToJSON;
