// streaming-availability/dist/models/ImageSet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageSetToJSON = exports.ImageSetFromJSONTyped = exports.ImageSetFromJSON = exports.instanceOfImageSet = void 0;
/**
 * Check if a given object implements the ImageSet interface.
 */
function instanceOfImageSet(value) {
    var isInstance = true;
    isInstance = isInstance && "lightThemeImage" in value;
    isInstance = isInstance && "darkThemeImage" in value;
    isInstance = isInstance && "whiteImage" in value;
    return isInstance;
}
exports.instanceOfImageSet = instanceOfImageSet;
function ImageSetFromJSON(json) {
    return ImageSetFromJSONTyped(json, false);
}
exports.ImageSetFromJSON = ImageSetFromJSON;
function ImageSetFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'lightThemeImage': json['lightThemeImage'],
        'darkThemeImage': json['darkThemeImage'],
        'whiteImage': json['whiteImage'],
    };
}
exports.ImageSetFromJSONTyped = ImageSetFromJSONTyped;
function ImageSetToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'lightThemeImage': value.lightThemeImage,
        'darkThemeImage': value.darkThemeImage,
        'whiteImage': value.whiteImage,
    };
}
exports.ImageSetToJSON = ImageSetToJSON;
