// streaming-availability/dist/models/NotFoundResponseSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NotFoundResponseSchemaToJSON = exports.NotFoundResponseSchemaFromJSONTyped = exports.NotFoundResponseSchemaFromJSON = exports.instanceOfNotFoundResponseSchema = void 0;
/**
 * Check if a given object implements the NotFoundResponseSchema interface.
 */
function instanceOfNotFoundResponseSchema(value) {
    var isInstance = true;
    isInstance = isInstance && "message" in value;
    return isInstance;
}
exports.instanceOfNotFoundResponseSchema = instanceOfNotFoundResponseSchema;
function NotFoundResponseSchemaFromJSON(json) {
    return NotFoundResponseSchemaFromJSONTyped(json, false);
}
exports.NotFoundResponseSchemaFromJSON = NotFoundResponseSchemaFromJSON;
function NotFoundResponseSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'message': json['message'],
    };
}
exports.NotFoundResponseSchemaFromJSONTyped = NotFoundResponseSchemaFromJSONTyped;
function NotFoundResponseSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'message': value.message,
    };
}
exports.NotFoundResponseSchemaToJSON = NotFoundResponseSchemaToJSON;
