// streaming-availability/dist/models/SearchFiltersResponseSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchFiltersResponseSchemaToJSON = exports.SearchFiltersResponseSchemaFromJSONTyped = exports.SearchFiltersResponseSchemaFromJSON = exports.instanceOfSearchFiltersResponseSchema = void 0;
var Show_1 = require("./Show");
/**
 * Check if a given object implements the SearchFiltersResponseSchema interface.
 */
function instanceOfSearchFiltersResponseSchema(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    isInstance = isInstance && "hasMore" in value;
    isInstance = isInstance && "nextCursor" in value;
    return isInstance;
}
exports.instanceOfSearchFiltersResponseSchema = instanceOfSearchFiltersResponseSchema;
function SearchFiltersResponseSchemaFromJSON(json) {
    return SearchFiltersResponseSchemaFromJSONTyped(json, false);
}
exports.SearchFiltersResponseSchemaFromJSON = SearchFiltersResponseSchemaFromJSON;
function SearchFiltersResponseSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (json['result'].map(Show_1.ShowFromJSON)),
        'hasMore': json['hasMore'],
        'nextCursor': json['nextCursor'],
    };
}
exports.SearchFiltersResponseSchemaFromJSONTyped = SearchFiltersResponseSchemaFromJSONTyped;
function SearchFiltersResponseSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (value.result.map(Show_1.ShowToJSON)),
        'hasMore': value.hasMore,
        'nextCursor': value.nextCursor,
    };
}
exports.SearchFiltersResponseSchemaToJSON = SearchFiltersResponseSchemaToJSON;
