// streaming-availability/dist/models/SearchTitleResponseSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchTitleResponseSchemaToJSON = exports.SearchTitleResponseSchemaFromJSONTyped = exports.SearchTitleResponseSchemaFromJSON = exports.instanceOfSearchTitleResponseSchema = void 0;
var Show_1 = require("./Show");
/**
 * Check if a given object implements the SearchTitleResponseSchema interface.
 */
function instanceOfSearchTitleResponseSchema(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfSearchTitleResponseSchema = instanceOfSearchTitleResponseSchema;
function SearchTitleResponseSchemaFromJSON(json) {
    return SearchTitleResponseSchemaFromJSONTyped(json, false);
}
exports.SearchTitleResponseSchemaFromJSON = SearchTitleResponseSchemaFromJSON;
function SearchTitleResponseSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': (json['result'].map(Show_1.ShowFromJSON)),
    };
}
exports.SearchTitleResponseSchemaFromJSONTyped = SearchTitleResponseSchemaFromJSONTyped;
function SearchTitleResponseSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': (value.result.map(Show_1.ShowToJSON)),
    };
}
exports.SearchTitleResponseSchemaToJSON = SearchTitleResponseSchemaToJSON;
