// streaming-availability/dist/models/SeriesStatus.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SeriesStatusToJSON = exports.SeriesStatusFromJSONTyped = exports.SeriesStatusFromJSON = exports.instanceOfSeriesStatus = exports.SeriesStatusStatusTextEnum = void 0;
/**
 * @export
 */
exports.SeriesStatusStatusTextEnum = {
    ReturningSeries: 'Returning Series',
    Planned: 'Planned',
    InProduction: 'In Production',
    Ended: 'Ended',
    Cancelled: 'Cancelled',
    Pilot: 'Pilot'
};
/**
 * Check if a given object implements the SeriesStatus interface.
 */
function instanceOfSeriesStatus(value) {
    var isInstance = true;
    isInstance = isInstance && "statusCode" in value;
    isInstance = isInstance && "statusText" in value;
    return isInstance;
}
exports.instanceOfSeriesStatus = instanceOfSeriesStatus;
function SeriesStatusFromJSON(json) {
    return SeriesStatusFromJSONTyped(json, false);
}
exports.SeriesStatusFromJSON = SeriesStatusFromJSON;
function SeriesStatusFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'statusCode': json['statusCode'],
        'statusText': json['statusText'],
    };
}
exports.SeriesStatusFromJSONTyped = SeriesStatusFromJSONTyped;
function SeriesStatusToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'statusCode': value.statusCode,
        'statusText': value.statusText,
    };
}
exports.SeriesStatusToJSON = SeriesStatusToJSON;
