// streaming-availability/dist/models/ServiceCountryInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceCountryInfoToJSON = exports.ServiceCountryInfoFromJSONTyped = exports.ServiceCountryInfoFromJSON = exports.instanceOfServiceCountryInfo = void 0;
var runtime_1 = require("../runtime");
var Addon_1 = require("./Addon");
var SupportedStreamingTypes_1 = require("./SupportedStreamingTypes");
/**
 * Check if a given object implements the ServiceCountryInfo interface.
 */
function instanceOfServiceCountryInfo(value) {
    var isInstance = true;
    isInstance = isInstance && "supportedStreamingTypes" in value;
    isInstance = isInstance && "addons" in value;
    return isInstance;
}
exports.instanceOfServiceCountryInfo = instanceOfServiceCountryInfo;
function ServiceCountryInfoFromJSON(json) {
    return ServiceCountryInfoFromJSONTyped(json, false);
}
exports.ServiceCountryInfoFromJSON = ServiceCountryInfoFromJSON;
function ServiceCountryInfoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'supportedStreamingTypes': (0, SupportedStreamingTypes_1.SupportedStreamingTypesFromJSON)(json['supportedStreamingTypes']),
        'addons': ((0, runtime_1.mapValues)(json['addons'], Addon_1.AddonFromJSON)),
    };
}
exports.ServiceCountryInfoFromJSONTyped = ServiceCountryInfoFromJSONTyped;
function ServiceCountryInfoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'supportedStreamingTypes': (0, SupportedStreamingTypes_1.SupportedStreamingTypesToJSON)(value.supportedStreamingTypes),
        'addons': ((0, runtime_1.mapValues)(value.addons, Addon_1.AddonToJSON)),
    };
}
exports.ServiceCountryInfoToJSON = ServiceCountryInfoToJSON;
