// streaming-availability/dist/models/ServicesResponseSchema.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServicesResponseSchemaToJSON = exports.ServicesResponseSchemaFromJSONTyped = exports.ServicesResponseSchemaFromJSON = exports.instanceOfServicesResponseSchema = void 0;
var runtime_1 = require("../runtime");
var DeprecatedServiceInfo_1 = require("./DeprecatedServiceInfo");
/**
 * Check if a given object implements the ServicesResponseSchema interface.
 */
function instanceOfServicesResponseSchema(value) {
    var isInstance = true;
    isInstance = isInstance && "result" in value;
    return isInstance;
}
exports.instanceOfServicesResponseSchema = instanceOfServicesResponseSchema;
function ServicesResponseSchemaFromJSON(json) {
    return ServicesResponseSchemaFromJSONTyped(json, false);
}
exports.ServicesResponseSchemaFromJSON = ServicesResponseSchemaFromJSON;
function ServicesResponseSchemaFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'result': ((0, runtime_1.mapValues)(json['result'], DeprecatedServiceInfo_1.DeprecatedServiceInfoFromJSON)),
    };
}
exports.ServicesResponseSchemaFromJSONTyped = ServicesResponseSchemaFromJSONTyped;
function ServicesResponseSchemaToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'result': ((0, runtime_1.mapValues)(value.result, DeprecatedServiceInfo_1.DeprecatedServiceInfoToJSON)),
    };
}
exports.ServicesResponseSchemaToJSON = ServicesResponseSchemaToJSON;
