// streaming-availability/dist/models/StreamingType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamingTypeToJSON = exports.StreamingTypeFromJSONTyped = exports.StreamingTypeFromJSON = exports.StreamingType = void 0;
/**
 * Type of the streaming option.
 * @export
 */
exports.StreamingType = {
    Free: 'free',
    Subscription: 'subscription',
    Buy: 'buy',
    Rent: 'rent',
    Addon: 'addon'
};
function StreamingTypeFromJSON(json) {
    return StreamingTypeFromJSONTyped(json, false);
}
exports.StreamingTypeFromJSON = StreamingTypeFromJSON;
function StreamingTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.StreamingTypeFromJSONTyped = StreamingTypeFromJSONTyped;
function StreamingTypeToJSON(value) {
    return value;
}
exports.StreamingTypeToJSON = StreamingTypeToJSON;
