// streaming-availability/dist/models/Subtitle.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubtitleToJSON = exports.SubtitleFromJSONTyped = exports.SubtitleFromJSON = exports.instanceOfSubtitle = void 0;
var runtime_1 = require("../runtime");
var Locale_1 = require("./Locale");
/**
 * Check if a given object implements the Subtitle interface.
 */
function instanceOfSubtitle(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfSubtitle = instanceOfSubtitle;
function SubtitleFromJSON(json) {
    return SubtitleFromJSONTyped(json, false);
}
exports.SubtitleFromJSON = SubtitleFromJSON;
function SubtitleFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'closedCaptions': !(0, runtime_1.exists)(json, 'closedCaptions') ? undefined : json['closedCaptions'],
        'locale': !(0, runtime_1.exists)(json, 'locale') ? undefined : (0, Locale_1.LocaleFromJSON)(json['locale']),
    };
}
exports.SubtitleFromJSONTyped = SubtitleFromJSONTyped;
function SubtitleToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'closedCaptions': value.closedCaptions,
        'locale': (0, Locale_1.LocaleToJSON)(value.locale),
    };
}
exports.SubtitleToJSON = SubtitleToJSON;
