// streaming-availability/dist/models/SupportedStreamingTypes.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 59 countries!
 *
 * The version of the OpenAPI document: 3.4.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SupportedStreamingTypesToJSON = exports.SupportedStreamingTypesFromJSONTyped = exports.SupportedStreamingTypesFromJSON = exports.instanceOfSupportedStreamingTypes = void 0;
/**
 * Check if a given object implements the SupportedStreamingTypes interface.
 */
function instanceOfSupportedStreamingTypes(value) {
    var isInstance = true;
    isInstance = isInstance && "addon" in value;
    isInstance = isInstance && "buy" in value;
    isInstance = isInstance && "rent" in value;
    isInstance = isInstance && "free" in value;
    isInstance = isInstance && "subscription" in value;
    return isInstance;
}
exports.instanceOfSupportedStreamingTypes = instanceOfSupportedStreamingTypes;
function SupportedStreamingTypesFromJSON(json) {
    return SupportedStreamingTypesFromJSONTyped(json, false);
}
exports.SupportedStreamingTypesFromJSON = SupportedStreamingTypesFromJSON;
function SupportedStreamingTypesFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'addon': json['addon'],
        'buy': json['buy'],
        'rent': json['rent'],
        'free': json['free'],
        'subscription': json['subscription'],
    };
}
exports.SupportedStreamingTypesFromJSONTyped = SupportedStreamingTypesFromJSONTyped;
function SupportedStreamingTypesToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'addon': value.addon,
        'buy': value.buy,
        'rent': value.rent,
        'free': value.free,
        'subscription': value.subscription,
    };
}
exports.SupportedStreamingTypesToJSON = SupportedStreamingTypesToJSON;
